src/server/game/Spells/SpellEffects.cpp

```diff
@@ -1742,6 +1742,86 @@ void Spell::EffectEnergize(SpellEffIndex effIndex)
     if (!unitTarget->IsAlive())
         return;
 
+    // Handle Secondary Potion Effects here
+    // Mad Alchemist's Potion (34440)
+    if (m_spellInfo->Id == 45051)
+    {
+        std::set<uint32> availableElixirs;
```

Comment:
Yes, there are hooks to which you can attach your functions, for example you can use an AfterCast hook. If in doubt, search for other script uses to see how it works, but you need to put this in your register subroutine:

```c++
void Register() override
{
    AfterCast += SpellCastFn(yourscriptclassname::yourfunction);
}
```